File: app/src/main/java/cz/kutner/comicsdb/MainActivity.java
package cz.kutner.comicsdb;

import android.app.SearchManager;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarActivity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.support.v7.widget.Toolbar;
import android.view.ViewGroup;
import android.widget.SearchView;

import java.util.ArrayList;
import java.util.List;

import cz.kutner.comicsdbclient.comicsdbclient.R;

public class MainActivity extends ActionBarActivity {
    public final static String COMICS_URL = "cz.kutner.comicsdbclient.comicsdbclient.comics_url";
    public final static String SEARCH_QUERY = "cz.kutner.comicsdbclient.comicsdbclient.search_query";
    private final String LOG_TAG = MainActivity.class.getSimpleName();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.container, new ComicsListFragment())
                    .commit();
        }
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);

        SearchManager searchManager =
                (SearchManager) getSystemService(Context.SEARCH_SERVICE);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        SearchView searchView =
                (SearchView) toolbar.findViewById(R.id.searchView);
        ComponentName cn = new ComponentName(this, ComicsSearchActivity.class);
        searchView.setSearchableInfo(searchManager.getSearchableInfo(cn));

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_about) {
            Intent intent = new Intent(this, AboutActivity.class);
            this.startActivity(intent);
        }

        return super.onOptionsItemSelected(item);
    }

    public void listItemOnClick(View v) {
        Intent intent = new Intent(this, ComicsDetailActivity.class);
        intent.putExtra(COMICS_URL, v.getTag().toString());
        startActivity(intent);
    }

}

File: app/src/main/kotlin/cz/kutner/comicsdb/activity/MainActivity.kt
package cz.kutner.comicsdb.activity

import android.app.SearchManager
import android.content.ComponentName
import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.view.GravityCompat
import android.support.v4.view.LayoutInflaterCompat
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.widget.SearchView
import com.mikepenz.iconics.IconicsDrawable
import com.mikepenz.iconics.context.IconicsLayoutInflater
import com.mikepenz.material_design_iconic_typeface_library.MaterialDesignIconic
import cz.kutner.comicsdb.R
import cz.kutner.comicsdb.fragment.*
import kotlinx.android.synthetic.main.activity.*
import kotlinx.android.synthetic.main.toolbar.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        LayoutInflaterCompat.setFactory(layoutInflater, IconicsLayoutInflater(delegate))
        super.onCreate(savedInstanceState)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().add(R.id.container, ComicsListFragment.newInstance()).commit()
        }
        setContentView(R.layout.activity)
        setupToolbar()
        setDrawerIcons()
        setActionsForDrawer()
    }

    private fun setupToolbar() {
        setSupportActionBar(toolbar)
        if (toolbar != null) {
            supportActionBar?.setDisplayHomeAsUpEnabled(true)
            toolbar.navigationIcon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_menu).color(Color.WHITE).sizeDp(24)
            toolbar.setNavigationOnClickListener { drawer_layout.openDrawer(GravityCompat.START) }
        }
    }

    private fun setDrawerIcons() {
        val menu = navigation_view.menu
        menu.findItem(R.id.navigation_item_comics).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_book_image).sizeDp(24)
        menu.findItem(R.id.navigation_item_news).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_comment_alert).sizeDp(24)
        menu.findItem(R.id.navigation_item_series).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_format_list_bulleted).sizeDp(24)
        menu.findItem(R.id.navigation_item_author).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_edit).sizeDp(24)
        menu.findItem(R.id.navigation_item_classified).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_money).sizeDp(24)
        menu.findItem(R.id.navigation_item_forum).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_comments).sizeDp(24)
        menu.findItem(R.id.navigation_item_about).icon = IconicsDrawable(this).icon(MaterialDesignIconic.Icon.gmi_info).sizeDp(24)
    }

    private fun setActionsForDrawer() {
        navigation_view.setNavigationItemSelectedListener { menuItem ->
            val fragment: Fragment = when (menuItem.itemId) {
                R.id.navigation_item_comics -> ComicsListFragment.newInstance()
                R.id.navigation_item_news -> NewsFragment.newInstance()
                R.id.navigation_item_series -> SeriesFragment.newInstance()
                R.id.navigation_item_author -> AuthorFragment.newInstance()
                R.id.navigation_item_classified -> ClassifiedFragment.newInstance()
                R.id.navigation_item_forum -> ForumFragment.newInstance()
                R.id.navigation_item_about -> AboutFragment.newInstance()
                else -> ComicsListFragment.newInstance()
            }
            menuItem.isChecked = true
            supportFragmentManager.beginTransaction().replace(R.id.container, fragment).commit()
            drawer_layout.closeDrawers()
            true
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.options_menu, menu)
        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
        val cn = ComponentName(this, SearchActivity::class.java)
        val searchView = menu.findItem(R.id.searchView).actionView as SearchView
        searchView.setSearchableInfo(searchManager.getSearchableInfo(cn))
        return true
    }

    companion object {
        val COMICS_ID: String = "cz.kutner.comicsdbclient.comicsdbclient.comics_id"
        val AUTHOR_ID: String = "cz.kutner.comicsdbclient.comicsdbclient.author_id"
        val SERIES_ID: String = "cz.kutner.comicsdbclient.comicsdbclient.series_id"
    }
}
